ruby/neo4j/driver/internal/internal_session.rb

```diff
@@ -48,11 +48,11 @@ def write_transaction(**config, &block)
       end
 
       def execute_read(**config, &block)
-        transaction(AccessMode::READ, **config, &block)
+        transaction(AccessMode::READ, **config, &(->(tx) { block.call(DelegatingTransaction.new(tx)) }))
```

Comment:
We have a nicer more idiomatic syntax for that,
```suggestion
        transaction(AccessMode::READ, **config) { block.call(DelegatingTransaction.new(it)) }
```
Alternatively composition,  bonus points for investigation. Personally never used it before:
```suggestion
        transaction(AccessMode::READ, **config, &(DelegatingTransaction.method(:new) >> block))
```
Looking at it I don't think it beats the former basic ruby syntax.

Reply:
I went with the first option - it's simple and readable. I also tested the second (with >>), and it works well too, thanks for the suggestions